src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1011,40 +1011,48 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
 
     if (intrin.op1 != nullptr)
     {
-        bool simdRegToSimdRegMove = false;
-
-        if ((intrin.id == NI_Vector64_CreateScalarUnsafe) || (intrin.id == NI_Vector128_CreateScalarUnsafe))
-        {
-            simdRegToSimdRegMove = varTypeIsFloating(intrin.op1);
-        }
-        else if (intrin.id == NI_AdvSimd_Arm64_DuplicateToVector64)
+        // Do not give a register to the first operand if it is contained.
+        if (intrin.op1->IsVectorZero() && !hasImmediateOperand && intrin.op1->isContained())
         {
-            simdRegToSimdRegMove = (intrin.op1->TypeGet() == TYP_DOUBLE);
+            assert(HWIntrinsicInfo::SupportsContainment(intrin.id));
         }
-        else if ((intrin.id == NI_Vector64_ToScalar) || (intrin.id == NI_Vector128_ToScalar))
+        else
         {
-            simdRegToSimdRegMove = varTypeIsFloating(intrinsicTree);
-        }
+            bool simdRegToSimdRegMove = false;
 
-        // If we have an RMW intrinsic or an intrinsic with simple move semantic between two SIMD registers,
-        // we want to preference op1Reg to the target if op1 is not contained.
-        if (isRMW || simdRegToSimdRegMove)
-        {
-            tgtPrefOp1 = !intrin.op1->isContained();
-        }
+            if ((intrin.id == NI_Vector64_CreateScalarUnsafe) || (intrin.id == NI_Vector128_CreateScalarUnsafe))
+            {
+                simdRegToSimdRegMove = varTypeIsFloating(intrin.op1);
+            }
+            else if (intrin.id == NI_AdvSimd_Arm64_DuplicateToVector64)
+            {
+                simdRegToSimdRegMove = (intrin.op1->TypeGet() == TYP_DOUBLE);
+            }
+            else if ((intrin.id == NI_Vector64_ToScalar) || (intrin.id == NI_Vector128_ToScalar))
+            {
+                simdRegToSimdRegMove = varTypeIsFloating(intrinsicTree);
+            }
 
-        if (intrinsicTree->OperIsMemoryLoadOrStore())
-        {
-            srcCount += BuildAddrUses(intrin.op1);
-        }
-        else if (tgtPrefOp1)
-        {
-            tgtPrefUse = BuildUse(intrin.op1);
-            srcCount++;
-        }
-        else
-        {
-            srcCount += BuildOperandUses(intrin.op1);
+            // If we have an RMW intrinsic or an intrinsic with simple move semantic between two SIMD registers,
+            // we want to preference op1Reg to the target if op1 is not contained.
+            if (isRMW || simdRegToSimdRegMove)
+            {
+                tgtPrefOp1 = !intrin.op1->isContained();
+            }
+
+            if (intrinsicTree->OperIsMemoryLoadOrStore())
+            {
+                srcCount += BuildAddrUses(intrin.op1);
+            }
+            else if (tgtPrefOp1)
+            {
+                tgtPrefUse = BuildUse(intrin.op1);
+                srcCount++;
+            }
+            else
+            {
+                srcCount += BuildOperandUses(intrin.op1);
```

Comment:
Why do you need to change this function? Did you an assert when running this?

Reply:
The diff looks a little confusing as @tannergooding pointed out, the original logic hasn't changed - all I did was add:
```cpp
if (intrin.op1->IsVectorZero() && !hasImmediateOperand && intrin.op1->isContained())
{
    assert(HWIntrinsicInfo::SupportsContainment(intrin.id));
}
```

Reply:
Basically, we don't want to get a register if the first operand is contained with a vector zero.